server/graphql/utils/request.ts

```diff
@@ -20,19 +22,60 @@ const getRequestsById = (requestIds, dataSources) => {
   return requestIds.map(id => dataSources.requests.getById(id))
 }
 
-const updateRequestHelper = (request, dataSources): Promise<ServerResponseInterface> => {
+const updateRequestHelper = async (request, dataSources): Promise<Document> => {
   if(!request.id) {
     throw new UserInputError('Missing argument value', { argumentName: 'id' })
   }
-  const requestTypeId = dataSources.requests.getById(request.id.toString()).requestType
-  return dataSources.requests.update(request)
-    .then(res => {
-      updateRequestTypeHelper({"id": requestTypeId}, dataSources)
-      return res
-    })
-    .catch(error => {
-      return error
-    })
+  const session = await mongoose.startSession()
+  try {
+    session.startTransaction()
+    const currentRequest = dataSources.requests.getById(request.id.toString())
+    const oldRequestTypeId = currentRequest.requestType.toString()
+    if(request.fulfilled === true && currentRequest.fulfilled === false) {
+      request.dateFulfilled = Date.now()
+    }
+    const res = await dataSources.requests.update(request, session)
+    if(request.requestType) {
+      const newRequestTypeId = request.requestType.toString()
+      const oldRequestType = dataSources.requestTypes.getById(oldRequestTypeId)
+      const newRequestType = dataSources.requestTypes.getById(request.requestType.toString())
+      oldRequestType.requests = oldRequestType.requests.filter(id => !id.equals(request.id))
+      newRequestType.requests.push(request.id)
+      await updateRequestTypeHelper({"id": oldRequestTypeId, "requests": oldRequestType.requests}, dataSources)
```

Comment:
It may be easier if you go right to calling a Mongoose update method with $push onto requestType.requests.